entity-management-mongo/src/main/java/eu/europeana/entitymanagement/mongo/config/DataSourceConfig.java

```diff
@@ -64,6 +78,43 @@ public MongoClient mongoClient() {
         (ConnectionPoolSettings.Builder builder) ->
             builder.maxConnectionIdleTime(mongoMaxIdleTimeMillisec, TimeUnit.MILLISECONDS);
 
+    try {
+      // build SSL context that uses separate truststore (needs to be copied and specified via
+      // mongo.truststore property)
+      if (!truststorePath.isBlank()) {
+        File tsFile =
+            ResourceUtils.getFile("classpath:" + truststorePath); // throws error if not available
+
+        KeyStore ks = KeyStore.getInstance("JKS");
+        ks.load(new FileInputStream(tsFile), truststorePwd.toCharArray());
```

Comment:
I recently discovered that when deploying as spring boot, the FileInputStream doesn't work properly (there are also some stakeoverflow issues reported). Therefore, I would suggest to use the File object only for verifying its existence (and throwing an error), and use the regular resourceAsInputStream to actually read the file from the jar
https://github.com/europeana/api-commons/commit/d61d9f1e7cccae8df994abfda90ff4de2d6caaf3

Reply:
I tested by running locally in IntelliJ and by deploying to CF test. In both situations it worked fine, so I'm not sure to what situation you are referring to. Can you elaborate?

Reply:
when the .war files are deplyoed to tomcat, the archive is extracted (exploded) and the file exists on the machine, but when running from the jar, the file is not exploded, and the content from the file included in the jar, needs to be read with the inputStram connected to the Resource (not the file),. See here: https://github.com/spring-projects/spring-boot/issues/9089

Reply:
land also this: https://stackoverflow.com/questions/44399422/read-file-from-resources-folder-in-spring-boot#comment91138175_49468282

Reply:
we will move the trustore outside the jar, so the implementation is ok